src/components/auth/LoginWithAuthServerPage.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useRef, useState } from 'react'
+import { ProviderType } from '@dcl/schemas'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { AuthServerProvider, connection } from 'decentraland-connect'
+import { authenticate } from '../../kernel-loader'
+import './LoginWithAuthServerPage.css'
+
+enum View {
+  WELCOME,
+  SIGN_IN_CODE,
+  EXPIRED
+}
+
+export const LoginWithAuthServerPage = () => {
+  const [view, setView] = useState<View>(View.WELCOME)
+  const [disabled, setDisabled] = useState(false)
+
+  const initSignInResultRef = useRef<any>()
+  const expirationTimeoutRef = useRef<NodeJS.Timeout>()
+
+  useEffect(() => {
+    const url = new URL(window.location.href)
+
+    switch (url.origin) {
+      case 'https://decentraland.org':
+        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.org')
+        AuthServerProvider.setAuthDappUrl('https://decentraland.org/auth')
+        break
+      case 'https://decentraland.today':
+        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.today')
+        AuthServerProvider.setAuthDappUrl('https://decentraland.today/auth')
+        break
+      default:
+        AuthServerProvider.setAuthServerUrl('https://auth-api.decentraland.zone')
+        AuthServerProvider.setAuthDappUrl('https://decentraland.zone/auth')
+    }
+
+    return () => {
+      clearTimeout(expirationTimeoutRef.current)
+    }
+  }, [])
+
+  useEffect(() => {
+    setDisabled(false)
+  }, [view])
+
+  if (view === View.WELCOME) {
+    return (
+      <div className="LoginWithAuthServerPage">
+        <div className="background"></div>
+        <div className="main">
+          <div className="logo"></div>
+          <div className="title">Discover a virtual social world</div>
+          <div className="subtitle">shaped by its community of creators & explorers.</div>
+          <Button
+            disabled={disabled}
+            className="button"
+            primary
+            onClick={async () => {
```

Comment:
In order to prevent a re-render of the Button and to have the code a bit more organized, would you mind rewritting this function using useCallback at the start of the component?
The same goes for other functions in buttons.